### PR TITLE
Small auth improvements

### LINKED ISSUE #429: add an option to bypass authentication in pyrog-client
when running pyrog (client and/or server) locally, we don't want to setup hydra, therefore we need an option to bypass/disable authentication.

we already have this in pyrog-server (ENV="test") but it could use a small refacto (use a AUTH_DISABLED env var like fhir-api ?)

### LINKED ISSUE #462: Raise error message when email does not exist
## Problem
The wheel keeps spinning when the connection failed if the email is null.

## Description
Error from server: 
```
Invalid options provided to ApolloServer: 
Invalid `prisma.user.upsert()` invocation in
/srv/dist/utils.js:114:55
{
  where: {
    email: null
           ~~~~
  },
  create: {
    email: null,
           ~~~~
    name: 'John'
  },
  update: {
    name: 'John'
  }
}
Argument email for where.email must not be null. Please use undefined instead.
Argument email: Got invalid value null on prisma.upsertOneUser. Provided null, expected String.

```

Thanks 🥨

### PR DESCRIPTION
**Pull request by Jasopaum**
Fixes #429
Fixes #462

### CODE REVIEW

File: client/src/constants/index.ts [hunk 1]
```diff
@@ -3,6 +3,11 @@ export const REFRESH_TOKEN_STORAGE_KEY = 'ARKHN_REFRESH_TOKEN';
 export const ID_TOKEN_STORAGE_KEY = 'ARKHN_ID_TOKEN';
 export const TOKEN_DATA_STORAGE_KEY = 'ARKHN_TOKEN_DATA';
 export const STATE_STORAGE_KEY = 'ARKHN_AUTH_STATE';
+export const AUTH_DISABLED =
+  process.env.REACT_APP_AUTH_DISABLED &&
```
Comment (simonvadee) [inline client/src/constants/index.ts:7]: do you need the `process.env.REACT_APP_AUTH_DISABLED &&` if you use the question mark syntax ?

File: server/src/server.ts [hunk 2]
```diff
@@ -39,7 +39,7 @@ if (IN_PROD) {
     async error => {
       const originalRequest = error.config
 
-      if (IN_PROD && error.response.status === 401 && !originalRequest._retry) {
+      if (error.response.status === 401 && !originalRequest._retry) {
```
Comment (simonvadee) [inline server/src/server.ts:42]: why removing this ?

File: server/src/permissions/index.ts [hunk 3]
```diff
@@ -17,24 +18,28 @@ export const authorizationError = {
 // We want to skip authorization checks during integration tests
 const skipAuthorizations = ENV === 'test'
 
+const checkUser = (user: User | undefined): Error | undefined => {
+  if (!user) {
+    return new Error(
+      `${authenticationError.code}: ${authenticationError.message}`,
+    )
+  }
+  if (!user.email) {
```
Comment (simonvadee) [inline server/src/permissions/index.ts:27]: I don't understand when this happen ? how can a user.email be null if user is not ?

File: client/src/constants/index.ts (hunk 1 above)
Comment (Jasopaum) [reply client/src/constants/index.ts:7]: Hum, right, it'll be better!

File: server/src/server.ts (hunk 2 above)
Comment (Jasopaum) [reply server/src/server.ts:42]: We're already in a `if (IN_PROD)`...

File: server/src/permissions/index.ts (hunk 3 above)
Comment (Jasopaum) [reply server/src/permissions/index.ts:27]: This happens when the idp doesn't provide an email for the user

File: client/src/constants/index.ts (hunk 1 above)
Comment (Jasopaum) [reply client/src/constants/index.ts:7]: Oh, the argument of `includes` can't be undefined... I can still remove the question mark though

Comment (simonvadee) [reply client/src/constants/index.ts:7]: right, it's fine :+1:

File: server/src/permissions/index.ts (hunk 3 above)
Comment (simonvadee) [reply server/src/permissions/index.ts:27]: right

**Review by simonvadee — APPROVED** [review]

**Pull request merged**